system/include/emscripten/em_js.h

```diff
@@ -50,6 +54,11 @@
 // We use <::> to separate the arguments from the function body because it isn't
 // valid anywhere in a C function declaration.
 
+// The __em_js_ref_foo pointer simply exists in order to force `foo` to be
+// undefined in the object file, even if there are no other local uses.  This
```

Comment:
Undefined? Should this be defined instead?

Reply:
EM_JS symbols are "undefined " in that they declare imports from the outside world.   So in this case the function `foo` is undefined and any usage of it creates a "undefined symbol" in the symbol table.

However, if there is no usage of it, it won't end up in the symbol table at all.   Which is a problem, because its the symbol table entry that hold the `import_name` attribute, which then tells the linker not to generate an error about it.

This change just creates a fake/unused usage of the symbol so it that it exists in the symbol table.

Reply:
Makes sense. But I was also confused by "undefined" here though so perhaps that could be reworded? Or we could add `(it is an import, so it is an undefined symbol; and we need to take its address so other compilation units can access it)` or such.

Reply:
Well do

Reply:
Done